File: dotnet/windntrees.net/DataAccess/DBFModels/Metadata/Product/LicenseInfo.cs
using SharedLibrary.Attributes;
using System.ComponentModel.DataAnnotations;

namespace DataAccess
{
    [MetadataType(typeof(LicenseInfoMD))]
    public partial class LicenseInfo
    {

    }

    public partial class LicenseInfoMD
    {
        [Key]
        [LocaleMessageRequired]
        [Display(ResourceType = typeof(LocaleResources.Contents.Product.LicenseInfo), Name = "ProductID")]
        public System.Guid ProductID { get; set; }

        [LocaleMessageRequired]
        [Display(ResourceType = typeof(LocaleResources.Contents.Product.LicenseInfo), Name = "Code")]
        public string Code { get; set; }

        [Timestamp]
        public byte[] RowVersion { get; set; }
    }
}

File: dotnet/windntrees.core/Application.Core/wwwroot/js/models/LicenseInfo.js
function LicenseInfo(data) {
    var instance = this;

    instance._datakey = data.ProductId;
    instance.ProductId = data.ProductId;

    instance.Code = ko.observable(data.Code);
    instance.Code.extend({
        maxLength: 1024
    });

    instance.RowVersion = data.RowVersion;
}